src/prefect/serialization/environment.py

```diff
@@ -13,17 +13,22 @@ class BaseEnvironmentSchema(ObjectSchema):
     class Meta:
         object_class = Environment
 
+    labels = fields.List(fields.String())
```

Comment:
If desired, these could be deserialized as a true `set` with a minimal custom field that wraps the `List` field:

```python
class Set(fields.List):
    def _deserialize(self, *args, **kwargs):
        return set(super()._deserialize(*args, **kwargs))

labels = Set(fields.String())
```

Reply:
Based on my deserialization tests, this appears to pass through the `__init__` method and is converted to a set at deserialization time anyway.  If you think the custom field approach is more robust though, I'm happy to make the change

Reply:
Oh I forgot about that -- totally fine then. Honestly I was looking at these tests when this occurred to me - https://github.com/PrefectHQ/prefect/pull/1442/files#diff-874215115fb45a0a8c40afed1ef73108R128 - I'm guessing the `set()` call is not needed then?

Reply:
Oops never mind these are serialization tests!

Reply:
Good to go!